h/traversal/roots.py

```diff
@@ -145,8 +145,15 @@ def __getitem__(self, client_id):
 class BulkAPIRoot(RootFactory):
     """Root factory for the Bulk API."""
 
-    # Currently only LMS uses this end-point
-    __acl__ = [(Allow, "client_authority:lms.hypothes.is", "bulk_action")]
+    def __acl__(self):
+        """Return ACL for bulk end-points."""
+
+        if authority := client_authority(self.request):
+            # Currently only LMS uses this end-point
+            if authority.startswith("lms.") and authority.endswith(".hypothes.is"):
```

Comment:
Should this use the `LMS_AUTHORITY_PATTERN` envvar?

Reply:
I've removed the "upsert" permission stuff meaning there isn't an `LMS_AUTHORITY_PATTERN` any more, so I think this solves both issues.